### PR TITLE
General: added fallback for broken ffprobe return

### PR DESCRIPTION
**Pull request by kalisp**
## Changelog Description
Customer provided .exr returned width and height equal to 0 which caused error in `extract_thumbnail`. This tries to use oiiotool to get metadata about file, in our case it read it correctly.

## Additional info
AnyDesk_XwH276p3SU

This PR is a bit weird as it solves very specific use case, which could be reoccurring though as source `.exr` comes from Deadline render from Maya and it is not supposed to be special. This update provide just one more layer of safety.

## Testing notes:
1. test on customer site OR
2. use attached test file from Clickup issue and run this in `Tray > Console`:
```
from openpype.lib.transcoding import  get_rescaled_command_arguments

input_path = "FILE_PATH"
args = get_rescaled_command_arguments("oiiotool", input_path, 1900, 1200)
print(args )
```

### CODE REVIEW

**Review by 64qam — APPROVED** [review]

File: openpype/lib/transcoding.py [hunk 1]
```diff
@@ -1345,6 +1341,40 @@ def get_rescaled_command_arguments(
...
+    # ffmpeg command
+    input_file_metadata = get_ffprobe_data(input_path, logger=log)
+    stream = input_file_metadata["streams"][0]
+    input_width = int(stream["width"])
+    input_height = int(stream["height"])
+
+    # fallback for weird files with width=0, height=0
+    if (input_width == 0 or input_height == 0) and application == "oiiotool":
+        # Load info about file from oiio tool
+        input_info = get_oiio_info_for_input(input_path, logger=log)
```
Comment (iLLiCiTiT) [inline openpype/lib/transcoding.py:1365]: Oiio is not always supported!

Comment (iLLiCiTiT) [reply openpype/lib/transcoding.py:1365]: I would say the issue is that it was always looking for first stream. There should be loop over all of them, use only those with `"video"` `"code_type"`.

File: openpype/lib/transcoding.py [hunk 2]
```diff
@@ -1345,6 +1341,40 @@ def get_rescaled_command_arguments(
...
+    Returns:
+        (tuple) (int, int, dict) - (height, width, sample_aspect_ratio)
+    Raises:
+        RuntimeError if image dimensions couldn't be parsed out.
+    """
+    # ffmpeg command
+    input_file_metadata = get_ffprobe_data(input_path, logger=log)
+    stream = input_file_metadata["streams"][0]
+    input_width = int(stream["width"])
+    input_height = int(stream["height"])
```
Comment (iLLiCiTiT) [inline openpype/lib/transcoding.py:1360]:
```suggestion
    input_width = input_height = 0
    stream = next(
        (
            s for s in input_file_metadata["streams"]
            if s.get("codec_type") == "video"
        ),
        None
    )
    if stream:
        input_width = int(stream["width"])
        input_height = int(stream["height"])
```

File: openpype/lib/transcoding.py [hunk 3]
```diff
@@ -1345,6 +1341,40 @@ def get_rescaled_command_arguments(
...
+        if input_info:
+            input_width = int(input_info["width"])
+            input_height = int(input_info["height"])
+
+    if input_width == 0 or input_height == 0:
+        raise RuntimeError("Couldn't read {} either "
+                           "with ffprobe or oiiotool".format(input_path))
+
+    stream_input_par = stream.get("sample_aspect_ratio")
+    return input_height, input_width, stream_input_par
```
Comment (iLLiCiTiT) [inline openpype/lib/transcoding.py:1375]:
```suggestion

    input_par = stream.get("sample_aspect_ratio") if stream else None
    if input_par:
        l_par, r_par = stream_input_par.split(":")
        input_par = float(l_par) / float(r_par)

    stream_input_par = stream.get("sample_aspect_ratio")
    return input_height, input_width, stream_input_par
```

File: openpype/lib/transcoding.py [hunk 4]
```diff
@@ -1345,6 +1341,40 @@ def get_rescaled_command_arguments(
...
+        input_info = get_oiio_info_for_input(input_path, logger=log)
+        if input_info:
+            input_width = int(input_info["width"])
+            input_height = int(input_info["height"])
+
+    if input_width == 0 or input_height == 0:
+        raise RuntimeError("Couldn't read {} either "
+                           "with ffprobe or oiiotool".format(input_path))
+
+    stream_input_par = stream.get("sample_aspect_ratio")
```
Comment (iLLiCiTiT) [inline openpype/lib/transcoding.py:1374]:
```suggestion
    stream_input_par = stream.get("sample_aspect_ratio") if stream else None
```

**Review by iLLiCiTiT — APPROVED** [review]

**Pull request merged**